browser/ui/webui/new_tab_page/brave_new_tab_page_handler.cc

```diff
@@ -73,30 +73,55 @@ void BraveNewTabPageHandler::ChooseLocalCustomBackground() {
 
 void BraveNewTabPageHandler::UseBraveBackground() {
   // Call ntp custom background images service.
-  profile_->GetPrefs()->SetBoolean(kNewTabPageCustomBackgroundEnabled, false);
+  auto* prefs = profile_->GetPrefs();
+  prefs->SetBoolean(kNewTabPageCustomBackgroundEnabled, false);
+  prefs->SetBoolean(kNewTabPageSolidColorBackgroundEnabled, false);
   OnCustomBackgroundImageUpdated();
   DeleteSanitizedImageFile();
 }
 
-bool BraveNewTabPageHandler::IsCustomBackgroundEnabled() const {
+void BraveNewTabPageHandler::UseSolidColorBackground(const std::string& color) {
+  auto* prefs = profile_->GetPrefs();
+  prefs->SetBoolean(kNewTabPageCustomBackgroundEnabled, false);
+  prefs->SetBoolean(kNewTabPageSolidColorBackgroundEnabled, true);
+  prefs->SetString(kNewTabPageSelectedSolidColorBackground, color);
+  OnCustomBackgroundImageUpdated();
+  DeleteSanitizedImageFile();
+}
+
+bool BraveNewTabPageHandler::IsCustomBackgroundImageEnabled() const {
   auto* prefs = profile_->GetPrefs();
   if (prefs->IsManagedPreference(prefs::kNtpCustomBackgroundDict))
     return false;
 
   return prefs->GetBoolean(kNewTabPageCustomBackgroundEnabled);
 }
 
+bool BraveNewTabPageHandler::IsSolidColorBackgroundEnabled() const {
+  auto* prefs = profile_->GetPrefs();
+  if (prefs->IsManagedPreference(prefs::kNtpCustomBackgroundDict))
+    return false;
+
+  return prefs->GetBoolean(kNewTabPageSolidColorBackgroundEnabled);
+}
+
 void BraveNewTabPageHandler::OnCustomBackgroundImageUpdated() {
   brave_new_tab_page::mojom::CustomBackgroundPtr value =
       brave_new_tab_page::mojom::CustomBackground::New();
   // Pass empty struct when custom background is disabled.
-  if (IsCustomBackgroundEnabled()) {
+  if (IsCustomBackgroundImageEnabled()) {
     // Add a timestamp to the url to prevent the browser from using a cached
     // version when "Upload an image" is used multiple times.
     std::string time_string = std::to_string(base::Time::Now().ToTimeT());
     std::string local_string(ntp_background_images::kCustomWallpaperURL);
     value->url = GURL(local_string + "?ts=" + time_string);
+  } else if (IsSolidColorBackgroundEnabled()) {
+    value->solid_color = profile_->GetPrefs()->GetString(
+        kNewTabPageSelectedSolidColorBackground);
   }
+
+  // TODO(sangwoo.ko) This doesn't notify other open NTP that background has
```

Comment:
When I implemented custom background images, only updating current NTP was intended behavior for showing sponsored image more longer :)

Reply:
Removed comment.